### DISCUSSION TITLE
Mui autocomplete expected object, received null

### DISCUSSION THREAD

**Discussion by DanPresa** [react-hook-form/react-hook-form#9267] [Q&A]
Hello there. I'm so sorry if this question is in other question.

I'm trying to `validate a MUI v5 autocomplete` element with a custom message but. I don't know how to do it.

I'm using `zod` to create an `schema` and try to validate it and also. I'm using controller element from `react-hook-form`.

As you know MUI autocomplete receive an object with `{label: '', value: ''}` properties. I need to get both properties to send them to an Api like and object. I only know how to validate strings like `textfields` I've looked for how to validate them but I wasn't able to find something.

When I try to validate it always throws a message like `expected object, received null`, null because I'm initilize it with a default value `null`.

```
const exampleSchema = z.object({
  name: z.string(),
  status: z.object({
    label: z.string(),
    value: z.string()
  })
})

type TypeValues = z.TypeOf<typeof exampleSchema>

const defaultValues: TypeValues = {
  name: '',
  status: null
}

const method = useForm<TypeValues>({
  resolver: zodResolver(),
  defaultValues,
})
```

### Please help me!!!

**Comment by leapful** [comment]
@DanPresa : Please provide a codesandbox url to re-produce the issue.

**Reply by DanPresa** [reply]
Sure thing. Here you are: https://codesandbox.io/s/validate-mui-autocomplete-zod-m6drzd

**Reply by leapful** [reply]
@DanPresa 

The codesandbox is showing validation error when auto complete has no value, it's a correct behavior, I don't see any error "expected object, received null".

**Reply by DanPresa** [reply]
I don't know how to show a custom error message.
I mean. I want to show a message like "This field is required"

**Reply by DanPresa** [reply]
@leapful I just change zod for yup.
I was able to validate and  show a custom error message for multiple fields.
Thanks for your time.
